iowrappers/photos_client.go

```diff
@@ -23,13 +28,14 @@ var isValidPhotoUrl = func(url string) bool {
 type PhotoURL string
 
 type PhotoClient interface {
-	GetPhotoURL(string) PhotoURL
+	GetPhotoURL(context.Context, string) PhotoURL
```

Comment:
Suggest return `(PhotoURL, error)`

Reply:
I think it might be okay to return PhotoURL. Since most of errors come from the place does not have valid image data, we can handle the exception by returning image of a default image.

Reply:
Previously I had trouble sending a default photo, because we don't have a file server to host such as default picture. Now it should be easier, since we encode the image to some binary string and embed them html response. I will add this default photo logic in a following PR.

Reply:
Feel free to let me know what you think. We can also handle this error on the caller side.

Reply:
> Previously I had trouble sending a default photo, because we don't have a file server to host such as default picture. Now it should be easier, since we encode the image to some binary string and embed them html response. I will add this default photo logic in a following PR.

We do have a file server to host images, in `/assets` directory

Reply:
Somehow it does not work simply by adding a local file to /assets folder. We will need to think about some other option to load a default picture.

Reply:
We eventually want to catch and handle the exception somewhere. I think the current implementation (returning PhotoURL instead of (PhotoURL, error)) is okay, because we need to return something in case Google maps API fails to provide valid data. The exception handling logic is done inside GetPhotoURL().

> Right, since the function `GetPhotoURL` is making an external HTTP call, it should return an error. The caller of this function should handle that error.

Reply:
The exception handling should not be done in this function, the caller should check the error and use a default URL if we do not have mitigation yet.

Reply:
Done, see [4d92fba](https://github.com/timwangmusic/Vacation-Planner/pull/338/commits/4d92fbacbfcac0d0847f62a05acaa06e0f8143d6)